cmd/minishift/cmd/openshift/registry.go

```diff
@@ -46,7 +47,7 @@ var registryCmd = &cobra.Command{
 
 		util.ExitIfNotRunning(host.Driver, constants.MachineName)
 
-		registryInfo, err := openshift.GetDockerRegistryInfo()
+		registryInfo, err := openshift.GetDockerRegistryInfo(addon.GetAddOnManager())
```

Comment:
I think the problem lies here (regarding circular dependencies and all). Why do you pass the AddOnManager? This creates an odd dependency in the openshift package to add-ons. All you want to know is whether the route add-on is enabled, right? So why don't just pass a bool and do the _isRegistryRouteEnabled_ here in the command.